pkg/apis/infinispan/v1/infinispan_types.go

```diff
@@ -49,6 +49,24 @@ type InfinispanServiceContainerSpec struct {
 	StorageClassName string  `json:"storageClassName,optional,omitempty"`
 }
 
+// UpgradePolicy specifies all the possible policies that rule the ispn cluster upgrade
```

Comment:
It would bi nice to add `+kubebuilder:validation:Enum` annotation for all possible values